Asterisk.2013/Asterisk.NET/Manager/ManagerConnection.cs

```diff
@@ -1106,11 +1121,29 @@ private void internalEventHandler(object sender, ManagerEvent e)
 					#endregion
 
 					case 62:
-						if (ConnectionState != null)
-						{
-							ConnectionState(this, (ConnectionStateEvent)e);
-							return;
-						}
+						if (Connect != null)

```

Comment:
This looks wrong! Reading the code, if Connect wasn't null, but the event was a ReloadEvent, the reload would never get hit, and you would instead send a connect event?